libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2230,13 +2230,17 @@ void GCS_MAVLINK::send_gps_global_origin() const
  */
 void GCS_MAVLINK::send_heartbeat() const
 {
+    MAV_STATE mav_state = system_status();
```

Comment:
shouldn't this be in system_status() function ?

Reply:
Yeah, probably.  We don't use it for anything else, however, and because I want this to win (for the time being) over whatever the vehicles wanted, I put it in like this.

Hmmm.

I think done properly you need to do a "greater of any of these values" sort of thing.  That's just going to be annoying, and a large patch.

I guess what you're after, then, is for the subclasses to have their methods renamed to `vehicle_system_status()` and we implement a "system_status" method on the base class?

Reply:
Separated status for vehicle and system could be a good thing. But that is outside the scope of this pr. Let's merge as it and implementation later!

Reply:
Changed this to put all the logic in the sytem_status call.  Yes, it is messier, but what we decided to do on the devcall.